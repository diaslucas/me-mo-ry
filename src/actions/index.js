import {
  FLIP_CARD, SET_CURRENT_PLAYER, SET_LAST_CARD_FLIPPED, ADD_POINT,
  SET_PLAYER_NAME, TOGGLE_SOUND, RESET_GAME } from '../constants';

export const flipCard = index => ({
  type: FLIP_CARD,
  payload: index,
});

export const setCurrentPlayer = id => ({
  type: SET_CURRENT_PLAYER,
  payload: id,
});

export const setLastCardFlipped = (card, index) => ({
  type: SET_LAST_CARD_FLIPPED,
  payload: { card, index },
});

export const addPoint = playerId => ({
  type: ADD_POINT,
  payload: playerId,
});

export const setPlayerName = (playerId, name) => ({
  type: SET_PLAYER_NAME,
  payload: { playerId, name },
});

export const toggleSound = () => ({
  type: TOGGLE_SOUND,
});

export const resetGame = () => ({
  type: RESET_GAME,
});
