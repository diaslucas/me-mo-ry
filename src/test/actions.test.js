/* eslint-disable no-undef */
import { flipCard, setLastCardFlipped, setCurrentPlayer, addPoint, setPlayerName } from '../actions';

describe('Testing Actions', () => {
  describe('Cards Actions', () => {
    test('should handle flipCard', () => {
      expect(flipCard(1)).toMatchSnapshot();
    });
  });

  describe('Game Actions', () => {
    test('should handle setLastCardFlipped', () => {
      expect(setLastCardFlipped({ id: 30, name: 'fish', color: 'pink', isFlipped: false }, 2)).toMatchSnapshot();
    });

    test('should handle setCurrentPlayer', () => {
      expect(setCurrentPlayer(1)).toMatchSnapshot();
    });
  });

  describe('Players Actions', () => {
    test('should handle addPoint', () => {
      expect(addPoint(1)).toMatchSnapshot();
    });

    test('should handle setPlayerName', () => {
      expect(setPlayerName(1, 'Lucas')).toMatchSnapshot();
    });
  });
});
