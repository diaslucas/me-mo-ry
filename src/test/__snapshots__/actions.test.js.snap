// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Actions Cards Actions should handle flipCard 1`] = `
Object {
  "payload": 1,
  "type": "FLIP_CARD",
}
`;

exports[`Testing Actions Game Actions should handle setCurrentPlayer 1`] = `
Object {
  "payload": 1,
  "type": "SET_CURRENT_PLAYER",
}
`;

exports[`Testing Actions Game Actions should handle setLastCardFlipped 1`] = `
Object {
  "payload": Object {
    "card": Object {
      "color": "pink",
      "id": 30,
      "isFlipped": false,
      "name": "fish",
    },
    "index": 2,
  },
  "type": "SET_LAST_CARD_FLIPPED",
}
`;

exports[`Testing Actions Players Actions should handle addPoint 1`] = `
Object {
  "payload": 1,
  "type": "ADD_POINT",
}
`;

exports[`Testing Actions Players Actions should handle setPlayerName 1`] = `
Object {
  "payload": Object {
    "name": "Lucas",
    "playerId": 1,
  },
  "type": "SET_PLAYER_NAME",
}
`;
